calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -336,6 +358,97 @@ def _apply_tags(self, profile_name, pod):
                 print('Could not create tag %s.\n%s' % (tag, e))
         print('Finished applying tags.')
 
+    def _get_annotations(self, pod):
```

Comment:
_get_annotations and _get_namespace look highly duplicated.

We could probably do with a generic "_get_pod_metadata" method which allows you to specify which bit of metadata you'd like.